Index: xfire-core/src/main/org/codehaus/xfire/handler/SoapHandler.java
===================================================================
RCS file: /home/projects/xfire/scm/xfire/xfire-core/src/main/org/codehaus/xfire/handler/SoapHandler.java,v
retrieving revision 1.15
diff -u -r1.15 SoapHandler.java
--- xfire-core/src/main/org/codehaus/xfire/handler/SoapHandler.java	5 Feb 2005 19:58:57 -0000	1.15
+++ xfire-core/src/main/org/codehaus/xfire/handler/SoapHandler.java	21 Feb 2005 16:11:36 -0000
@@ -46,6 +46,12 @@
         this.bodyHandler = bodyHandler;
     }
 
+    public SoapHandler( EndpointHandler bodyHandler, Handler headerHandler)
+    {
+        this.bodyHandler = bodyHandler;
+        this.headerHandler = headerHandler;
+    }
+
     /**
      * Invoke the Header and Body Handlers for the SOAP message.
      */
@@ -78,6 +84,8 @@
                 {
                     readHeaders(context);
                     validateHeaders(context);
+
+                    headerHandler.invoke(context);
                 }
                 else if ( reader.getLocalName().equals("Body") )
                 {
Index: xfire-spring/src/main/org/codehaus/xfire/spring/XFireExporter.java
===================================================================
RCS file: /home/projects/xfire/scm/xfire/xfire-spring/src/main/org/codehaus/xfire/spring/XFireExporter.java,v
retrieving revision 1.2
diff -u -r1.2 XFireExporter.java
--- xfire-spring/src/main/org/codehaus/xfire/spring/XFireExporter.java	20 Feb 2005 20:42:33 -0000	1.2
+++ xfire-spring/src/main/org/codehaus/xfire/spring/XFireExporter.java	21 Feb 2005 16:13:39 -0000
@@ -5,6 +5,7 @@
 
 import org.codehaus.xfire.XFire;
 import org.codehaus.xfire.handler.SoapHandler;
+import org.codehaus.xfire.handler.Handler;
 import org.codehaus.xfire.message.ObjectServiceHandler;
 import org.codehaus.xfire.service.object.DefaultObjectService;
 import org.codehaus.xfire.service.object.Invoker;
@@ -23,12 +24,12 @@
     extends RemoteExporter
     implements Controller, InitializingBean
 {
-    private BeanFactory factory;
     private DefaultObjectService service;
     private ServiceBuilder serviceBuilder;
     private SpringXFireController controller;
     private XFire xFire;
-    
+    private Handler headerHandler;
+
     public void afterPropertiesSet()
         throws Exception
     {
@@ -37,8 +38,8 @@
         getXfire().getServiceRegistry().register( service );
 
         Invoker invoker = new BeanInvoker(getProxyForService());
-        service.setServiceHandler(new SoapHandler(new ObjectServiceHandler(invoker)));
-        
+        service.setServiceHandler(new SoapHandler(new ObjectServiceHandler(invoker),headerHandler));
+
         controller = new SpringXFireController(getXfire(), service.getName());
     }
 
@@ -68,5 +69,15 @@
     public void setXfire(XFire xFire)
     {
         this.xFire = xFire;
+    }
+
+    public Handler getHeaderHandler()
+    {
+        return headerHandler;
+    }
+
+    public void setHeaderHandler(Handler headerHandler)
+    {
+        this.headerHandler = headerHandler;
     }
 }
